index.test-d.ts

```diff
@@ -56,3 +56,15 @@ ky(server.url, {searchParams: 'foo=bar'});
 ky(server.url, {searchParams: {foo: 'bar'}});
 ky(server.url, {searchParams: {foo: 1}});
 ky(server.url, {searchParams: new URLSearchParams({foo: 'bar'})});
+
+// `json` option
+ky(server.url, {
+	json: {
```

Comment:
nit: `GET` requests are not supposed to have a body, so while these tests serve their purpose, they are actually demonstrating improper usage.

Reply:
We could automatically set `method` to `POST` if user provides a body with no method.

Reply:
Hmm my initial reaction is that feels too magical to me. I actually didn't realize `got` did that. Who says they want `POST` instead of `PUT` or `PATCH`? I suppose we should align their behavior, though.

Reply:
> Who says they want POST instead of PUT or PATCH? 

If someone wants PUT or PATCH, then sets `method: 'PUT'` or `method: 'PATCH'` ;)

Reply:
Right, it's easy to be explicit about that, and personally, I would force the user to choose a method.

Reply:
Continued in https://github.com/sindresorhus/got/issues/639#issuecomment-455042713

Reply:
> GET requests are not supposed to have a body, so while these tests serve their purpose, they are actually demonstrating improper usage.

@sholladay Any ideas how we could enforce this in TS?

Reply:
My first thought is to use an overload, but there are some other approaches [detailed here](http://artsy.github.io/blog/2018/11/21/conditional-types-in-typescript/) that I haven't used and may be worth exploring.

Reply:
Continued in https://github.com/sindresorhus/ky/issues/85